sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -366,6 +369,60 @@ def test_model_pipeline_same_as_normalize_true(LinearModel, params):
     assert_allclose(y_pred_normalize, y_pred_standardize)
 
 
+# FIXME: 'normalize' to be removed in 1.2
+@pytest.mark.filterwarnings("ignore:'normalize' was deprecated")
+@pytest.mark.parametrize(
+    "estimator, is_sparse, with_mean",
+    [(LinearRegression, True, False),
+     (LinearRegression, False, True),
+     (LinearRegression, False, False)]
+)
+def test_linear_model_sample_weights_normalize_in_pipeline(
+        estimator, is_sparse, with_mean
+):
```

Comment:
If this test is only meant to test `LinearRegression` it should be moved to `sklearn/linear_model/tests/test_base.py`. If it's meant to be extended to `Ridge`, `Lasso`... maybe it should be move to a new file, e.g. `sklearn/linear_model/tests/test_linear_model.py`

Reply:
If I recall I was proposing `sklearn/linear_model/tests/test_common.py` that is the usual way that we structure common tests for a module.

Reply:
To anticipate this question, I tried to see if this test would pass with the current code for Ridge and Lasso and actually it always fails whether `with_mean` is `True` or `False` on dense data and it also fails with `with_mean=False` on sparse data:

```
sklearn/linear_model/tests/test_coordinate_descent.py::test_linear_model_sample_weights_normalize_in_pipeline[LinearRegression-True-False] PASSED                                                          [ 12%]
sklearn/linear_model/tests/test_coordinate_descent.py::test_linear_model_sample_weights_normalize_in_pipeline[LinearRegression-False-True] PASSED                                                          [ 25%]
sklearn/linear_model/tests/test_coordinate_descent.py::test_linear_model_sample_weights_normalize_in_pipeline[LinearRegression-False-False] PASSED                                                         [ 37%]
sklearn/linear_model/tests/test_coordinate_descent.py::test_linear_model_sample_weights_normalize_in_pipeline[Ridge-True-False] FAILED                                                                     [ 50%]
sklearn/linear_model/tests/test_coordinate_descent.py::test_linear_model_sample_weights_normalize_in_pipeline[Ridge-False-True] FAILED                                                                     [ 62%]
sklearn/linear_model/tests/test_coordinate_descent.py::test_linear_model_sample_weights_normalize_in_pipeline[Ridge-False-False] FAILED                                                                    [ 75%]
sklearn/linear_model/tests/test_coordinate_descent.py::test_linear_model_sample_weights_normalize_in_pipeline[Lasso-False-True] FAILED                                                                     [ 87%]
sklearn/linear_model/tests/test_coordinate_descent.py::test_linear_model_sample_weights_normalize_in_pipeline[Lasso-False-False] FAILED                                                                    [100%]
```

So the deprecation of their `normalize` option should not be implemented with the same message I believe.

To keep this PR focused, let's just move this test to `sklearn/linear_model/tests/test_base.py` for now.

Reply:
Yes @glemaitre . I answered your comment, but it must have gone lost int the flow of other comments:
https://github.com/scikit-learn/scikit-learn/pull/17743#discussion_r555219796

I will move it to `test_base.py`. 

@ogrisel failing for Ridge and Lasso might indeed be a problem as it was supposed to be extended to include them in this test. Why is this the case (for their failing)?

Reply:
@glemaitre
There is no directory: `sklearn/linear_model/tests/test_common.py`
(there is: `sklearn/tests/test_common.py`, hence my previous question above).

Should I create it?

Reply:
> There is no file: sklearn/linear_model/tests/test_common.py

We could create one. But it's fine to keep in sklearn/linear_model/tests/test_base.py for now.